### PR TITLE
KRV-23286: E2E changes to validate role/rolebindings removal

### PR DESCRIPTION
**Pull request by gilltaran**
# Description
Enhance the existing e2e test with a new step to validate role/rolebinding removal

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1221 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration
image

### CODE REVIEW

**Review by AkshaySainiDell — CHANGES_REQUESTED** [review]

File: tests/e2e/steps/steps_def.go [hunk 1]
```diff
@@ -1432,3 +1432,24 @@ func (step *Step) validateRbacCreated(_ Resource, namespace string) error {
 
 	return nil
 }
+
+func (step *Step) validateRbacDeleted(_ Resource, namespace string) error {
+	fmt.Println("validating Rbac deletion on namespace", namespace)
+
+	cmd := exec.Command("kubectl", "get", "rolebindings", "-n", "namespacename")
```
[the remark this answers is not in the document]
Comment (gilltaran) [reply tests/e2e/steps/steps_def.go]: addressed

**Review by AkshaySainiDell — DISMISSED** [review]
lgtm

**Review by HarishH-DELL — APPROVED** [review]

**Review by nitesh3108 — APPROVED** [review]

**Review by abhi16394 — APPROVED** [review]
LGTM

File: tests/e2e/testfiles/connectivity-values.yaml [hunk 2]
```diff
@@ -48,5 +48,6 @@
     - "Validate connectivity client from CR [1] is not installed"
     - "Enable forceRemoveDriver on CR [2]"
     - "Delete custom resource [2]"
+    - "Validate rbac deleted from namespace [test-vxflexos]"
```
Comment (alexemc) [inline tests/e2e/testfiles/connectivity-values.yaml:51]: I suggest placing this right after "Validate connectivity client from CR [1] is not installed" - that's when the role and rolebindings should be deleted.

Comment (alexemc) [inline tests/e2e/testfiles/connectivity-values.yaml:51]: Can we check all namespaces (not just `test-vxflexos`) for the presence of the connectivity client role/bindings?

**Review by alexemc — CHANGES_REQUESTED** [review]

File: tests/e2e/testfiles/connectivity-values.yaml (hunk 2 above)
Comment (ashleyvjoy) [reply tests/e2e/testfiles/connectivity-values.yaml:51]: Addressed here: https://github.com/dell/csm-operator/pull/632

Comment (ashleyvjoy) [reply tests/e2e/testfiles/connectivity-values.yaml:51]: Addressed here: https://github.com/dell/csm-operator/pull/632

**Pull request merged**